Assets/Scripts/Dungeon/DungeonGenerator.cs

```diff
@@ -7,35 +7,34 @@
 using UnityEditor;
 using UnityEngine;
 using UnityEngine.Tilemaps;
+using UnityEngine.UIElements;
 
 public class DungeonGenerator : MonoBehaviour
 {
     private List<Room> availableRooms = new List<Room>();
-    private List<Room> PlacedRooms = new List<Room>();
+    private List<Room> placedRooms = new List<Room>();
 
-    private Dictionary<Direction, List<Room>> RoomsByDirection = new Dictionary<Direction, List<Room>>();
+    private Dictionary<Direction, List<Room>> roomsByDirection = new Dictionary<Direction, List<Room>>();
     
-    public int count = 0;
-    public int roomCount = 50;
-    public int seed;
-    public int additionalDistance = 10;
-    public int maxOffset = 20;
-
-    public Tilemap tilemap;
-    public Tile corridorTile;
-    public Tile wallTile;
+    [SerializeField] private int count = 0;
+    [SerializeField] private int roomCount = 50;
+    [SerializeField] private int seed = 0;
```

Comment:
Perhaps init this with Rand.NextInt(). So if you do not want to change the seed manually that it will be different every new +addcomponent atleast.